submarine-workbench/interpreter/python-interpreter/src/test/java/org/apache/submarine/interpreter/PythonInterpreterTest.java

```diff
@@ -24,35 +24,99 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import static org.junit.Assert.assertEquals;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
-import java.io.IOException;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
 
 public class PythonInterpreterTest {
   private static final Logger LOG = LoggerFactory.getLogger(PythonInterpreterTest.class);
 
-  private static PythonInterpreter pythonInterpreter = null;
+  private static PythonInterpreter testPythonInterpreter1 = null;
+
+  private static PythonInterpreter testPythonInterpreter2 = null;
 
   @BeforeClass
   public static void setUp() {
-    pythonInterpreter = new PythonInterpreter();
-    pythonInterpreter.open();
+    testPythonInterpreter1 = new PythonInterpreter();
+    testPythonInterpreter2 = new PythonInterpreter();
+    testPythonInterpreter1.open();
+    testPythonInterpreter2.open();
   }
 
   @AfterClass
-  public static void tearDown() throws Exception {
-    if (null != pythonInterpreter) {
-      pythonInterpreter.close();
+  public static void tearDown()  {
+    if (null != testPythonInterpreter1) {
+      testPythonInterpreter1.close();
+    }
+    if (null != testPythonInterpreter2) {
+      testPythonInterpreter2.close();
     }
   }
 
+
   @Test
-  public void calcOnePlusOne() throws IOException {
+  public void calcOnePlusOne() {
     String code = "1+1";
-    InterpreterResult result = pythonInterpreter.interpret(code);
+    InterpreterResult result = testPythonInterpreter1.interpret(code);
     LOG.info("result = {}", result);
 
     assertEquals(result.code(), InterpreterResult.Code.SUCCESS);
     assertEquals(result.message().get(0).getData(), "2\n"); // 1 + 1 = 2 + '\n'
   }
+
+  private class infinityPythonJobforCancel implements Runnable {
+    @Override
+    public void run() {
+      String code = "import time\nwhile True:\n  time.sleep(1)";
```

Comment:
Why is `time.sleep(1)`?

Reply:
Thanks for the review, the python code is " while True: time.sleep(1)" which aims to simulate a long running code.